### ISSUE TITLE
ideas for improving output regarding "exceptions" in .nsprc

### ISSUE THREAD

**Issue by jokeyrhyme** [nodesecurity/nsp#190]
After setting up an exception for https://nodesecurity.io/advisories/525 today, it occurred to me that `nsp check` ought to behave just a little bit differently with regard to this:

-   when there are exceptions listed in .nsprc, rather than output "No known vulnerabilities found", perhaps it should say something like "These known vulnerabilities ignored: ..." with a list, and then "No other known vulnerabilities found"

    -  I think it's a good idea to remind anyone reading logs that exceptions have been made

-   when there is an exception listed in .nsprc that no longer applies, e.g. after we update tough-cookie to 2.3.3 or 2.4.0 or something, then `nsp check` should exit with a non-zero exit code (unless --warn-only is used) and display something like "This exception ... does not match any known vulnerabilities found"

    -  I think it's a good idea to force people to keep their exceptions list minimal, so they can detect when vulnerabilities creep back in

Is this something that you'd like to see a PR for? Any suggestions or comments?

**Comment by evilpacket** [comment]
Thx for the issue!

These are things we've talked about in the past and I tend to agree with you that we should remind people about things they didn't see rather than "it's all good".

I'll chat with the team  and figure out where this is best to implement this. It might end up having to be done in the API (which is currently a private repo) so that the cli can make a proper informed output.

**Comment by migg24** [comment]
Came here to report the same thing but it was already said. Something like

> "No known vulnerabilities found (3 ignored)"

would be completely fine with me.

**Comment by jokeyrhyme** [comment]
Yeah, I'm not sure I'd want the full tabular result output for ignored advisories: either the sum total as @migg24 suggests or a list of only the advisory URLs, short and sweet :)
